homeassistant/components/automation/state.py

```diff
@@ -21,6 +22,34 @@
 CONF_STATE = "state"
 CONF_FOR = "for"
 
+BASE_SCHEMA = vol.Schema({
+    vol.Required(CONF_PLATFORM): 'state',
+    vol.Required(CONF_ENTITY_ID): cv.entity_id,
+    # These are str on purpose. Want to catch YAML conversions
+    CONF_STATE: str,
+    CONF_FOR: vol.All(vol.Schema({
+        CONF_HOURS: vol.Coerce(int),
+        CONF_MINUTES: vol.Coerce(int),
+        CONF_SECONDS: vol.Coerce(int),
+    }), cv.has_at_least_one_key(CONF_HOURS, CONF_MINUTES, CONF_SECONDS)),
+})
+
+TRIGGER_SCHEMA = vol.Schema(vol.All(
+    BASE_SCHEMA.extend({
+        # These are str on purpose. Want to catch YAML conversions
+        CONF_FROM: str,
+        CONF_TO: str,
+    }),
+    vol.Any(cv.key_dependency(CONF_FOR, CONF_TO),
+            cv.key_dependency(CONF_FOR, CONF_STATE))
+))
+
+IF_ACTION_SCHEMA = vol.Schema(vol.All(
+    BASE_SCHEMA,
+    vol.Any(cv.key_dependency(CONF_FOR, CONF_TO),
```

Comment:
`IF_ACTION_SCHEMA` does not have a `CONF_TO` key so this part of the dependency should never be valid.